src/main/java/tc/oc/pgm/community/commands/ModerationCommands.java

```diff
@@ -0,0 +1,436 @@
+    if (warn) {
+      warn(sender, target, match, reason, true);
+    }
+
+    if (punish(PunishmentType.MUTE, targetMatchPlayer, sender, reason, silent)) {
+      chat.addMuted(targetMatchPlayer);
+    }
+  }
+
+  @Command(
+      aliases = {"unmute", "um"},
+      usage = "<player>",
+      desc = "Unmute a player",
+      perms = Permissions.MUTE)
+  public void unMute(CommandSender sender, Player target, Match match) {
+    MatchPlayer targetMatchPlayer = match.getPlayer(target);
+    if (chat.isMuted(targetMatchPlayer)) {
+      chat.removeMuted(targetMatchPlayer);
+
+      targetMatchPlayer.sendMessage(
+          new PersonalizedTranslatable("moderation.unmute.target")
+              .getPersonalizedText()
+              .color(ChatColor.GREEN));
+
+      sender.sendMessage(
+          new PersonalizedTranslatable(
+                  "moderation.unmute.sender", targetMatchPlayer.getStyledName(NameStyle.FANCY))
+              .color(ChatColor.GRAY));
+    } else {
+      sender.sendMessage(
+          new PersonalizedTranslatable(
+                  "moderation.unmute.none", targetMatchPlayer.getStyledName(NameStyle.FANCY))
+              .getPersonalizedText()
+              .color(ChatColor.RED));
+    }
+  }
+
+  @Command(
+      aliases = {"warn", "w"},
+      usage = "<player> <reason> -s (silent)",
+      desc = "Warn a player for bad behavior",
+      perms = Permissions.WARN)
+  public void warn(
+      CommandSender sender,
+      Player target,
+      Match match,
+      @Text String reason,
+      @Switch('s') boolean silent) {
+    MatchPlayer targetMatchPlayer = match.getPlayer(target);
+
+    if (punish(PunishmentType.WARN, targetMatchPlayer, sender, reason, silent)) {
+      sendWarning(targetMatchPlayer, reason);
+    }
+  }
+
+  @Command(
+      aliases = {"kick", "k"},
+      usage = "<player> <reason> -s (silent)",
+      desc = "Kick a player from the server",
+      perms = Permissions.KICK)
+  public void kick(
+      CommandSender sender,
+      Player target,
+      Match match,
+      @Text String reason,
+      @Switch('s') boolean silent) {
+    MatchPlayer targetMatchPlayer = match.getPlayer(target);
+    if (punish(PunishmentType.KICK, targetMatchPlayer, sender, reason, silent)) {
+      target.kickPlayer(
+          formatPunishmentScreen(
+              PunishmentType.KICK, formatPunisherName(sender, match), reason, null));
+    }
+  }
+
+  @Command(
+      aliases = {"ban", "permban", "pb"},
+      usage = "<player> <reason> -s (silent)",
+      desc = "Ban a player from the server forever",
+      perms = Permissions.BAN)
+  public void ban(
+      CommandSender sender,
+      Player target,
+      Match match,
+      @Text String reason,
+      @Switch('s') boolean silent) {
+    MatchPlayer targetMatchPlayer = match.getPlayer(target);
+    Component senderName = formatPunisherName(sender, match);
+    if (punish(PunishmentType.BAN, targetMatchPlayer, sender, reason, silent)) {
```

Comment:
This will punish the player with `Duration.ZERO`, shouldn't this be defined with `Duration.MAX_VALUE`?

Reply:
I should have made a comment regarding this, but from what I understand there is no ‘Duration.MAX_VALUE’. See [here](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html) 

But basically I’m using the logic so that: if a punishment has a duration then it must be able to expire, otherwise a zero value represents infinity.

Reply:
Ah ok, that works